phoenix-core/src/main/java/org/apache/phoenix/jdbc/PhoenixHAAdminTool.java

```diff
@@ -0,0 +1,634 @@
+     * @throws Exception when parsing or reading from the input file
+     */
+    @VisibleForTesting
+    List<ClusterRoleRecord> readRecordsFromFile(String file) throws Exception {
+        Preconditions.checkArgument(!StringUtils.isEmpty(file));
+        String fileType = FilenameUtils.getExtension(file);
+        switch (fileType) {
+        case "json":
+            // TODO: use jackson or standard JSON library according to PHOENIX-5789
+            try (Reader reader = new FileReader(file)) {
+                ClusterRoleRecord[] records =
+                        JacksonUtil.getObjectReader(ClusterRoleRecord[].class).readValue(reader);
+                return Arrays.asList(records);
+            }
+        case "yaml":
+            LOG.error("YAML file is not yet supported. See W-8274533");
+        default:
+            throw new Exception("Can not read cluster role records from file '" + file + "' " +
+                    "reason: unsupported file type");
+        }
+    }
+
+    /**
+     * Helper method to write the given cluster role records into the ZK clusters respectively.
+     *
+     * // TODO: add retry logics
+     *
+     * @param records The cluster role record list to save on ZK
+     * @param forceful if true, this method will ignore errors on other clusters; otherwise it will
+     *                 not update next cluster (in order) if there is any failure on current cluster
+     * @return a map of HA group name to list cluster's url for cluster role record failing to write
+     */
+    private Map<String, List<String>> syncClusterRoleRecords(List<ClusterRoleRecord> records,
+            boolean forceful) throws IOException {
+        Map<String, List<String>> failedHaGroups = new HashMap<>();
+        for (ClusterRoleRecord record : records) {
+            String haGroupName = record.getHaGroupName();
+            try (PhoenixHAAdminHelper admin1 = new PhoenixHAAdminHelper(record.getZk1(), getConf(), HighAvailibilityCuratorProvider.INSTANCE);
+                    PhoenixHAAdminHelper admin2 = new PhoenixHAAdminHelper(record.getZk2(), getConf(), HighAvailibilityCuratorProvider.INSTANCE)) {
+                // Update the cluster previously ACTIVE cluster first.
+                // It reduces the chances of split-brain between clients and clusters.
+                // If can not determine previous ACTIVE cluster, update new STANDBY cluster first.
+                final PairOfSameType<PhoenixHAAdminHelper> pair;
+                if (admin1.isCurrentActiveCluster(haGroupName)) {
+                    pair = new PairOfSameType<>(admin1, admin2);
+                } else if (admin2.isCurrentActiveCluster(haGroupName)) {
+                    pair = new PairOfSameType<>(admin2, admin1);
+                } else if (record.getRole(admin1.getZkUrl()) == ClusterRole.STANDBY) {
+                    pair = new PairOfSameType<>(admin1, admin2);
+                } else {
+                    pair = new PairOfSameType<>(admin2, admin1);
```

Comment:
how would we ever get here? (if neither 1 nor 2 are ACTIVE, and 1 isn't STANDBY?)

Reply:
In case cluster 1 is offline or unknown I believe.  In general we shouldn't see this state.